src/qt/bitcoin.cpp

```diff
@@ -552,15 +552,9 @@ int main(int argc, char *argv[])
 
     std::unique_ptr<interfaces::Node> node = interfaces::MakeNode();
 
-    /// 1. Parse command-line options. These take precedence over anything else.
-    // Command-line options take precedence:
-    node->setupServerArgs();
-    SetupUIArgs();
-    node->parseParameters(argc, argv);
-
     // Do not refer to data directory yet, this can be overridden by Intro::pickDataDirectory
 
-    /// 2. Basic Qt initialization (not dependent on parameters or configuration)
+    /// 1. Basic Qt initialization (not dependent on parameters or configuration)
```

Comment:
Are you sure this is safe?
I haven't tested, but following the code, `BitcoinApplication` constructor on L573 does use `gArgs.GetArg("-uiplatform"`.

Reply:
I'm not sure about unsafe, but it certainly is not good practice to check for whether the argument is set before actually parsing the arguments. This also makes it so that the `-uiplatform` option is effectively ignored. It appears that I have been misled by the comment on this line.

How would you recommend that an error be shown to the user if they don't set the correct arguments?

Reply:
Checking command line arguments in a constructor generally seems to be bad practice, as it becomes pretty hard to reason about ordering. Is there a way that can be avoided (by passing in the argument value into the constructor instead?).

Reply:
To get around this problem, I moved the platform style setup out of the constructor and set it up after `parseparameters`.